engine/session_live.js

```diff
@@ -792,13 +779,22 @@ class SessionLive {
       let pushPromises = [];
       for (let i = 0; i < Object.keys(this.mediaManifestURIs).length; i++) {
         let bw = Object.keys(this.mediaManifestURIs)[i];
-        pushPromises.push(this._parseMediaManifest(this.liveSourceM3Us[bw].M3U, bw, this.mediaManifestURIs[bw], bw));
-        //debug(`[${this.sessionId}]: Pushed pushPromise for bw=${bw}`);
+        pushPromises.push(this._parseMediaManifest(this.liveSourceM3Us[bw].M3U, this.mediaManifestURIs[bw], bw));
+        debug(`[${this.sessionId}]: Pushed pushPromise for bw=${bw}`);
       }
-
       // Segment Pushing
       debug(`[${this.sessionId}]: Executing Promises II: Segment Pushing`);
-      await Promise.all(pushPromises);
+      let results = await allSettled(pushPromises);//await Promise.all(pushPromises);
+      const removedDiscSeqList = results.map((item) => {
+        if (item.status === "rejected") {
+          return -1;
+        }
+        return item.value.removedDiscSeqs;
+      });
+      if (removedDiscSeqList.every((val, i, arr) => val === arr[0])) {
+        // if all variants removed equal amounts of disc-tags
+        this.discSeqCount += removedDiscSeqList[0];
+      }
```

Comment:
Earlier implementation, would increment `discSeqCount` per variant. 
This version will increment only after each variant has pushed & shifted items.